x-pack/plugins/spaces/server/lib/space_schema.ts

```diff
@@ -8,7 +8,7 @@ import Joi from 'joi';
 import { MAX_SPACE_INITIALS } from '../../common/constants';
 
 export const spaceSchema = Joi.object({
-  id: Joi.string().regex(/[a-z0-9_\-]*/, `lower case, a-z, 0-9, "_", and "-" are allowed`),
+  id: Joi.string().regex(/^[a-z0-9_\-]+$/, `lower case, a-z, 0-9, "_", and "-" are allowed`),
```

Comment:
__suggestion:__ would you mind adding a basic Jest unit test that would guard us against any accidentally made mistakes in this pattern in the future?

Reply:
Yup, you got it!